Sources/Sentry/SentryNSURLRequest.m

```diff
@@ -75,7 +75,7 @@ - (_Nullable instancetype)initStoreRequestWithDsn:(SentryDsn *)dsn
 - (_Nullable instancetype)initEnvelopeRequestWithDsn:(SentryDsn *)dsn
                                              andData:(NSData *)data
                                  didFailWithError:(NSError *_Nullable *_Nullable)error {
-    NSURL *apiURL = [dsn getStoreEndpoint];
+    NSURL *apiURL = [dsn getEnvelopeEndpoint];
```

Comment:
I know there aren't any tests for this right now, but would it be possible to add a simple test to check if this points to the correct endpoint?

Reply:
Added a test for each endpoint